tools/autograd/templates/VariableType.cpp

```diff
@@ -364,11 +364,11 @@ Tensor & VariableType::s_copy_(Tensor & self, const Tensor & src, bool async) co
   auto& src_ = unpack_any(src, "src", 1);
   check_inplace(self);
   std::shared_ptr<CopyBackwards> grad_fn;
-  auto requires_grad = compute_requires_grad({ self, src });
+  auto requires_grad = compute_requires_grad(self, src);
   requires_grad &= isFloatingPoint(self.type().scalarType());
   if (requires_grad) {
     grad_fn = std::make_shared<CopyBackwards>();
-    grad_fn->next_functions = compute_next_functions({ self, src });
+    grad_fn->next_functions = Function::flags(self, src).next_functions;
```

Comment:
Are we using `FunctionFlags` anywhere outside of autogenerated code? If its only purpose is computing `next_functions`, then we could remove `requires_grad` from it and save some work

Reply:
Yes, I noticed this when I inlined the definition of `next_functions`. I don't think you would save very much work splitting it up, since `requires_grad` is just a boolean. Not much wasted computation.

Reply:
It would still be cleaner to only compute the stuff we actually need